### ISSUE TITLE
No deprecation warning for method call on deprecated type alias

### ISSUE THREAD

**Issue by bluss** [rust-lang/rust#31848]
The following should emit a deprecation warning (`feature(deprecated)`)

``` rust
#![feature(deprecated)]
pub mod lib {
    #[deprecated(note="don't use this")]
    pub type Foo = String;
}

fn main() {
    let t = ::lib::Foo::new(); // no warning
    println!("{}", t);
}
```

Type aliases can emit warnings, like in the following example, but I think it should warn as the use site too (marked "no warning"):

``` rust
#![feature(deprecated)]
pub mod lib {
    #[deprecated(note="don't use this")]
    pub type Foo = String;
}

use lib::Foo; // WARNING use of deprecated item

fn main() {
    let s = Foo::new(); // no warning
    println!("{}", s);
}
```

**Comment by bluss** [comment]
cc @petrochenkov

**Comment by petrochenkov** [comment]
This is probably caused by the same reasons as https://github.com/rust-lang/rust/issues/30209

**Comment by Mark-Simulacrum** [comment]
Closing.

```
warning: use of deprecated item: don't use this
 --> test.rs:7:13
  |
7 |     let t = ::lib::Foo::new(); // no warning
  |             ^^^^^^^^^^^^^^^
  |
  = note: #[warn(deprecated)] on by default
```
